app/Repositories/Assignment/GenerateAssignmentFromTemplateRepository.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace App\Repositories\Assignment;
+
+use App\Models\Assignment;
+use App\Models\AssignmentTemplate;
+use Carbon\Carbon;
+use Illuminate\Database\Eloquent\Collection;
+
+class GenerateAssignmentFromTemplateRepository
+{
+  public function generate()
+  {
+    $templates = AssignmentTemplate::with(['days'])->where('enabled', '=', 1)->get();
+
+    $this->generateAssignmentsFromTemplates($templates);
+
+    return $templates;
+  }
+
+  private function generateAssignmentsFromTemplates(Collection $templates)
+  {
+    foreach ($templates as $template) {
+      $days = new Collection($template->days);
+      $this->generateAssignmentsFromTemplate($template, $days);
+    }
```

Comment:
`foreach ($templates as $template) {
      $this->generateAssignmentsFromTemplate($template, collect($template->days));
    }
`

Reply:
$template->days lo puede obtener la otra funcion a partir de template, no tendrias que mandarlo aca